internal/service/activation.go

```diff
@@ -0,0 +1,42 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/spacemeshos/explorer-backend/model"
+)
+
+// GetActivations returns atxs by filter.
+func (e *Service) GetActivations(ctx context.Context, page, perPage int64) (atxs []*model.Activation, total int64, err error) {
+	return e.getActivations(ctx, &bson.D{}, e.getFindOptions("id", page, perPage))
+}
+
+// GetActivation returns atx by id.
+func (e *Service) GetActivation(ctx context.Context, activationID string) (*model.Activation, error) {
+	filter := &bson.D{{"id", strings.ToLower(activationID)}}
+	atx, total, err := e.getActivations(ctx, filter, options.Find().SetLimit(1).SetProjection(bson.D{{"_id", 0}}))
+	if err != nil {
+		return nil, fmt.Errorf("error find atx: %w", err)
+	}
+	if total == 0 {
+		return nil, ErrNotFound
+	}
+	return atx[0], nil
+}
+
+func (e *Service) getActivations(ctx context.Context, filter *bson.D, options *options.FindOptions) (atxs []*model.Activation, total int64, err error) {
+	total, err = e.storage.CountActivations(ctx, filter)
```

Comment:
pity `storage` is not an interface but I guess that's in todo list to improve :)